torch/csrc/jit/codegen/cuda/root_domain_map.cpp

```diff
@@ -76,6 +80,11 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
   if (consumer_tv_->definition()->isA<TransposeOp>()) {
     return mapTranspose(
         producer, consumer, root_dims_to_map, producer_to_consumer);
+  } else if (
+      consumer_tv_->definition()->isA<TorchGatherOp>() &&
+      require_same_extent_) {
```

Comment:
Shouldn't this be limited to the dimension undergoing the gather? Shouldn't there also be logic for index_select to prevent mapping that dimension too?

Reply:
For index_select, AFAIK, all dimensions except for the selected one should have the same extent, so should be safe to map. The selected domain is excluded from the mapping.

Reply:
More of a conceptual nit than anything else. I don't like the idea of "same extent", as that to me bleeds into "exact map" and what I think what we're saying is the dimension being involved in select/gather/index_select just should be unmappable from producer->consumer. The reason we think it should be unmappable is because it doesn't simply support inlining, and some other properties we pointed out. We likely should tease out these properties more formally, but for now I'm happy with these changes.